src/slider.ts

```diff
@@ -0,0 +1,1331 @@
+import './label.js';
+import { html, LitElement } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { createRef, ref } from 'lit/directives/ref.js';
+import { customElement, property, state } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+import { when } from 'lit/directives/when.js';
+import packageJson from '../package.json' with { type: 'json' };
+import { LocalizeController } from './library/localize.js';
+import styles from './slider.styles.js';
+import type FormControl from './library/form-control.js';
+import shadowRootMode from './library/shadow-root-mode.js';
+import final from './library/final.js';
+import required from './library/required.js';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'glide-core-slider': Slider;
+  }
+}
+
+/**
+ * @attr {string} label
+ * @attr {boolean} [disabled=false]
+ * @attr {boolean} [hide-label=false]
+ * @attr {number} [max=100]
+ * @attr {number} [min=0]
+ * @attr {boolean} [multiple=false]
+ * @attr {string} [name='']
+ * @attr {'horizontal'|'vertical'} [orientation='horizontal']
+ * @attr {boolean} [readonly=false]
+ * @attr {boolean} [required=false]
+ * @attr {number} [step=1]
+ * @attr {string} [tooltip]
+ * @attr {number[]} [value=[]]
+ *
+ * @readonly
+ * @attr {string} [version]
+ *
+ * @slot {Element | string} [description] - Additional information or context
+ *
+ * @fires {Event} change
+ * @fires {Event} input
+ * @fires {Event} invalid
+ *
+ * @readonly
+ * @prop {HTMLFormElement | null} form
+ *
+ * @readonly
+ * @prop {ValidityState} validity
+ *
+ * @method checkValidity
+ * @returns boolean
+ *
+ * @method formAssociatedCallback
+ * @method formResetCallback
+ *
+ * @method reportValidity
+ * @returns boolean
+ *
+ * @method resetValidityFeedback
+ *
+ * @method setCustomValidity
+ * @param {string} message
+ *
+ * @method setValidity
+ * @param {ValidityStateFlags} [flags]
+ * @param {string} [message]
+ */
+@customElement('glide-core-slider')
+@final
+export default class Slider extends LitElement implements FormControl {
+  static formAssociated = true;
+
+  static override shadowRootOptions: ShadowRootInit = {
+    ...LitElement.shadowRootOptions,
+    mode: shadowRootMode,
+    delegatesFocus: true,
+  };
+
+  static override styles = styles;
+
+  @property({ reflect: true, useDefault: true })
+  name = '';
+
+  // Intentionally not reflected to match native.
+  /**
+   * @default []
+   */
+  @property({ type: Array })
+  get value(): number[] {
+    if (
+      this.multiple &&
+      this.minimumValue !== undefined &&
+      this.maximumValue !== undefined
```

Comment:
`!== undefined` rather than a truthy check, because we need to allow for `0`

Reply:
I was just working on lint stuff. How do you feel about [this](https://typescript-eslint.io/rules/strict-boolean-expressions) rule?

Reply:
I think I dig it. Would help to prevent foot guns, I think. How about you?

Reply:
Same, yeah. Plus it's explicit. It's just usually a hard sell to developers, who often aren't used to it.